clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -62,7 +65,8 @@ public ConsumerRecord(String topic,
 
 
     /**
-     * Creates a record to be received from a specified topic and partition
+     * Creates a record to be received from a specified topic and partition (provided for
+     * compatibility with Kafka 0.10 before the message format supported headers).
```

Comment:
This is inline with the comment left when the constructor for timestamp was added. It also is the same, it created a valid / correct message format still, and you can still use the constructor.